app.rb

```diff
@@ -5,138 +5,135 @@
+            JSON.pretty_generate(data: campaign, relationships: trackers)
+        else
+            halt 404, "PROJECT NOT FOUND: #{params[:id]}"
+        end
+    end
 
-    campaign = Campaign[params[:id]]
+    post '/api/v1/campaigns/?' do
+        begin
+            new_data = JSON.parse(request.body.read)
+            saved_campaign = Campaign.create(new_data)
+        rescue => e
+            logger.info "FAILED to create new campaign: #{e.inspect}"
+            halt 400
+        end
 
-    JSON.pretty_generate(data: campaign.trackers)
-  end
+        new_location = URI.join(@request_url.to_s + '/', saved_campaign.id.to_s).to_s
 
-  get '/api/v1/campaigns/:campaign_id/trackers/:id/?' do
-    content_type 'application/json'
+        status 201
+        headers('Location' => new_location)
+    end
 
-    begin
-      doc_url = URI.join(@request_url.to_s + '/', 'json')
-      tracker = Tracker
-                .where(campaign_id: params[:campaign_id], id: params[:id])
-                .first
-      halt(404, 'Tracker not found') unless tracker
-      JSON.pretty_generate(data: {
-                             tracker: tracker,
-                             links: { item_url: doc_url }
-                           })
-    rescue => e
-      status 400
-      logger.info "FAILED to process GET tracker request: #{e.inspect}"
-      e.inspect
+    get '/api/v1/campaigns/:id/trackers/?' do
+        content_type 'application/json'
+        campaign = Campaign.where(id: params[:id]).first
+        JSON.pretty_generate(data: campaign.trackers)
     end
-  end
-
-  get '/api/v1/campaigns/:campaign_id/trackers/:id/json' do
-    content_type 'text/plain'
-
-    begin
-      Tracker
-        .where(campaign_id: params[:campaign_id], id: params[:id])
-        .first
-        .json
-    rescue => e
-      status 404
-      e.inspect
+
+    get '/api/v1/campaigns/:campaign_id/trackers/:id/?' do
+        content_type 'application/json'
+
+        begin
+            doc_url = URI.join(@request_url.to_s + '/', 'json')
+            tracker = Tracker
+                      .where(campaign_id: params[:campaign_id], id: params[:id])
+                      .first
+            halt(404, 'Tracker not found') unless tracker
+            JSON.pretty_generate(data: {
+                                     tracker: tracker,
+                                     links: { item_url: doc_url }
+                                 })
+        rescue => e
+            status 400
+            logger.info "FAILED to process GET tracker request: #{e.inspect}"
+            e.inspect
+        end
     end
-  end
-
-  post '/api/v1/campaigns/:campaign_id/trackers/?' do
-    begin
-      new_data = JSON.parse(request.body.read)
-      campaign = Campaign[params[:campaign_id]]
-      saved_tracker = campaign.add_tracker(new_data)
-    rescue => e
-      logger.info "FAILED to create new tracker: #{e.inspect}"
-      halt 400
+
+    get '/api/v1/campaigns/:campaign_id/trackers/:id/json' do
```

Comment:
Do we need this route compare to that there is '/api/v1/campaigns/:campaign_id/trackers/:id/?' route